thorlcr/activities/indexread/thindexread.cpp

```diff
@@ -382,20 +398,21 @@ class CIndexCountActivityMaster : public CIndexReadBase
         CIndexReadBase::init();
         if (!container.queryLocalOrGrouped())
         {
-            if (helper->canMatchAny())
-                limit = (rowcount_t)helper->getChooseNLimit();
-            else
+            if (!helper->canMatchAny())
                 totalCountKnown = true; // totalCount = 0
```

Comment:
Doesn't this need to set it to 0, otherwise in a child query it may get the wrong answer on a second call?